MdePkg/Test/Mock/Library/GoogleTest/MockUefiBootServicesTableLib/MockUefiBootServicesTableLib.cpp

```diff
@@ -10,56 +10,143 @@ MOCK_INTERFACE_DEFINITION (MockUefiBootServicesTableLib);
+  gBS_RegisterProtocolNotify,                                                          // EFI_REGISTER_PROTOCOL_NOTIFY
+  gBS_LocateHandle,                                                                    // EFI_LOCATE_HANDLE
+  gBS_LocateDevicePath,                                                                // EFI_LOCATE_DEVICE_PATH
+  NULL,                                                                                // EFI_INSTALL_CONFIGURATION_TABLE
+  NULL,                                                                                // EFI_IMAGE_LOAD
+  NULL,                                                                                // EFI_IMAGE_START
+  NULL,                                                                                // EFI_EXIT
+  NULL,                                                                                // EFI_IMAGE_UNLOAD
+  NULL,                                                                                // EFI_EXIT_BOOT_SERVICES
+  NULL,                                                                                // EFI_GET_NEXT_MONOTONIC_COUNT
+  NULL,                                                                                // EFI_STALL
+  NULL,                                                                                // EFI_SET_WATCHDOG_TIMER
+  gBS_ConnectController,                                                               // EFI_CONNECT_CONTROLLER
+  gBS_DisconnectController,                                                            // EFI_DISCONNECT_CONTROLLER
+  gBS_OpenProtocol,                                                                    // EFI_OPEN_PROTOCOL
+  gBS_CloseProtocol,                                                                   // EFI_CLOSE_PROTOCOL
+  NULL,                                                                                // EFI_OPEN_PROTOCOL_INFORMATION
+  NULL,                                                                                // EFI_PROTOCOLS_PER_HANDLE
+  gBS_LocateHandleBuffer,                                                              // EFI_LOCATE_HANDLE_BUFFER
+  gBS_LocateProtocol,                                                                  // EFI_LOCATE_PROTOCOL
+  gBS_InstallMultipleProtocolInterfaces,                                               // EFI_INSTALL_MULTIPLE_PROTOCOL_INTERFACES
+  (EFI_UNINSTALL_MULTIPLE_PROTOCOL_INTERFACES)gBS_UninstallMultipleProtocolInterfaces, // EFI_UNINSTALL_MULTIPLE_PROTOCOL_INTERFACES
+  NULL,                                                                                // EFI_CALCULATE_CRC32
+  gBS_CopyMem,                                                                         // EFI_COPY_MEM
+  gBS_SetMem,                                                                          // EFI_SET_MEM
+  gBS_CreateEventEx                                                                    // EFI_CREATE_EVENT_EX
 };
 
 extern "C" {

```

Comment:
What about gST and gImageHandle?  What unit test code that tests code under test that uses those globals do?

Reply:
gImageHandle is set to NULL but its value can be manipulated in a test. 
I added mocks for prototols in gST and the other struct members can similarly be accessed and modified in any test.